src/workspace/subcomponents/_TabUtils.js

```diff
@@ -0,0 +1,182 @@
+ */
+const enableFocusStyles = (event) => {
+  event.currentTarget.setAttribute('data-focus-styles-enabled', 'true');
+};
+
+/**
+ * Disables focus styles for the target of the given event. Typically used as an onMouseDown callback on selectable elements.
+ */
+const disableFocusStyles = (event) => {
+  event.currentTarget.setAttribute('data-focus-styles-enabled', 'false');
+};
+
+/**
+ * Sets focus to the 'next' tab element.
+ * @param {object} event The triggering event.
+ * @param {number} index The index of the current element.
+ * @param {array} ids The array of id strings.
+ */
+const nextFocus = (event, index, ids) => {
+  event.preventDefault();
+  event.stopPropagation();
+
+  if (index < 0) {
+    const element = document.getElementById(ids[0]);
+    if (element) {
+      element.focus();
+    }
+    return;
+  }
+
+  const newIndex = index + 1 >= ids.length ? 0 : index + 1;
+  const element = document.getElementById(ids[newIndex]);
+  if (element) {
+    element.focus();
+  }
+};
+
+/**
+ * Sets focus to the 'previous' tab element.
+ * @param {object} event The triggering event.
+ * @param {number} index The index of the current element.
+ * @param {array} ids The array of id strings.
+ */
+const previousFocus = (event, index, ids) => {
+  event.preventDefault();
+  event.stopPropagation();
+
+  if (index < 0) {
+    const element = document.getElementById(ids[ids.length - 1]);
+    if (element) {
+      element.focus();
+    }
+    return;
+  }
+
+  const newIndex = index - 1 < 0 ? ids.length - 1 : index - 1;
+  const element = document.getElementById(ids[newIndex]);
+  if (element) {
+    element.focus();
+  }
+};
+
+/**
+ * Sets focus to the 'first' tab element.
+ * @param {object} event The triggering event.
+ * @param {number} index The index of the current element.
+ * @param {array} ids The array of id strings.
+ */
+const firstFocus = (event, index, ids) => {
+  event.preventDefault();
+  event.stopPropagation();
+
+  if (index !== 0) {
+    const element = document.getElementById(ids[0]);
+    if (element) {
+      element.focus();
+    }
+  }
+};
+
+/**
+ * Sets focus to the 'end' tab element.
+ * @param {object} event The triggering event.
+ * @param {number} index The index of the current element.
+ * @param {array} ids The array of id strings.
+ */
+const lastFocus = (event, index, ids) => {
+  event.preventDefault();
+  event.stopPropagation();
+
+  if (index !== ids.length - 1) {
+    const element = document.getElementById(ids[ids.length - 1]);
+    if (element) {
+      element.focus();
+    }
+  }
+};
+
+/**
+ * The default event handler for navigational arrow keys.
+ * @param {object} event The triggering event.
+ * @param {number} index The index of the current element.
+ * @param {array} ids The array of id strings.
+ */
+const handleArrows = (event, index, ids) => {
+  const isRTL = document.getElementsByTagName('html')[0].getAttribute('dir') === 'rtl';
```

Comment:
seems like we should have a common util for this or something... musing more than anything, you don' t have to change it.

Reply:
agreed